### PR TITLE
feat(anoncreds): support credential attribute value and marker

### PR DESCRIPTION
**Pull request by Vickysomtee**
The PR implements the support for queries by credential attribute presence and value

**work funded by the Government of Ontario**

### CODE REVIEW

File: packages/anoncreds/src/repository/AnonCredsCredentialRecord.ts [hunk 1]
```diff
@@ -12,24 +13,28 @@ export interface AnonCredsCredentialRecordProps {
...
+    [key: `attr::${string}::value`]: string | undefined
 }
 
 export type CustomAnonCredsCredentialTags = {
   schemaName: string
   schemaVersion: string
   schemaIssuerId: string
   issuerId: string
-}
+} & TagsBase
```
Comment (TimoGlastra) [inline packages/anoncreds/src/repository/AnonCredsCredentialRecord.ts:37]: This can be deleted again

```suggestion
}
```

**Review by TimoGlastra — COMMENTED** [review]
Implementation looks good, but we're missing some tags

**Review by genaris — COMMENTED** [review]
Pretty clean! I think it would be nice to have at least a few tests to make sure different border cases are covered.

**Comment by TimoGlastra** [comment]
> Pretty clean! I think it would be nice to have at least a few tests to make sure different border cases are covered.

That's what i meant instead of tags lol

File: packages/anoncreds/src/repository/AnonCredsCredentialRecord.ts [hunk 2]
```diff
@@ -62,15 +66,21 @@ export class AnonCredsCredentialRecord extends BaseRecord<
...
-    return {
+    const tags: Tags<DefaultAnonCredsCredentialTags, CustomAnonCredsCredentialTags> = {
       ...this._tags,
       credentialDefinitionId: this.credential.cred_def_id,
       schemaId: this.credential.schema_id,
       credentialId: this.credentialId,
       credentialRevocationId: this.credentialRevocationId,
-      revocationRegistryId: this.credential.rev_reg_id,
       linkSecretId: this.linkSecretId,
       attributes: Object.keys(this.credential.values),
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply packages/anoncreds/src/repository/AnonCredsCredentialRecord.ts:76]: Okay. In that case we can remove the `attributes` array

[the remark this answers is not in the document]
Comment (TimoGlastra) [reply packages/anoncreds/src/repository/AnonCredsCredentialRecord.ts:76]: Yes!

File: packages/anoncreds-rs/src/services/__tests__/AnonCredsRsHolderService.test.ts [hunk 3]
```diff
@@ -360,6 +364,19 @@ describe('AnonCredsRsHolderService', () => {
...
     })
 
+    test('referent with attribute values and marker restriction', async () => {
+      await anonCredsHolderService.getCredentialsForProofRequest(agentContext, {
+        proofRequest,
+        attributeReferent: 'attr5_referent',
+      })
+
+      expect(findByQueryMock).toHaveBeenCalledWith(agentContext, {
+        attributes: ['name'],
```
Comment (TimoGlastra) [inline packages/anoncreds-rs/src/services/__tests__/AnonCredsRsHolderService.test.ts:374]: This one should not be added anymore

Comment (TimoGlastra) [reply packages/anoncreds-rs/src/services/__tests__/AnonCredsRsHolderService.test.ts:374]: Not sure where it's coming from?

**Review by TimoGlastra — APPROVED** [review]

File: packages/anoncreds-rs/src/services/__tests__/AnonCredsRsHolderService.test.ts (hunk 3 above)
Comment (TimoGlastra) [reply packages/anoncreds-rs/src/services/__tests__/AnonCredsRsHolderService.test.ts]: Fixed

**Pull request merged**